### PR TITLE
Added validation mechanism for API docs generated from TypeDoc

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature (docs): Added validation mechanism for API docs generated from TypeDoc. Closes ckeditor/ckeditor5#12965.

Internal (typedoc-plugins): Added `event:` prefix for event names to make sure they are unique in the parent's scope.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

File: packages/ckeditor5-dev-docs/lib/validators/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+				errors.add( error );
+			} );
+
+			errors.forEach( error => typeDoc.logger.warn( error ) );
+		}
+
+		typeDoc.logger.info( 'Validation completed.' );
+
+		return result;
```
Comment (pomek) [inline packages/ckeditor5-dev-docs/lib/validators/index.js:47]:
```suggestion
		return errors.size === 0;
```
or
```suggestion
		return !errors.size;
```

Then, we can remove the `result` variable and simplify the `onError` callback.

File: packages/ckeditor5-dev-docs/lib/validators/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		for ( const validator of validators ) {
+			const errors = new Set();
+
+			validator( project, error => {
+				result = false;
+
+				errors.add( error );
+			} );
+
+			errors.forEach( error => typeDoc.logger.warn( error ) );
```
Comment (pomek) [inline packages/ckeditor5-dev-docs/lib/validators/index.js:42]: Is it possible to add a node from the source code where an error occurs? See: https://github.com/ckeditor/ckeditor5-dev/blob/7472b759bc6ccc380dc6016a63ec4a6da737a0ed/packages/typedoc-plugins/lib/tag-event/index.js#L75.

File: packages/ckeditor5-dev-docs/lib/validators/index.js (hunk 1 above)
Comment (psmyrek) [reply packages/ckeditor5-dev-docs/lib/validators/index.js]: I simplified the main loop, thanks.

File: packages/ckeditor5-dev-docs/lib/validators/index.js (hunk 2 above)
Comment (psmyrek) [reply packages/ckeditor5-dev-docs/lib/validators/index.js]: I already tried to do that, but it turns out that not all reflections have a symbol: `project.getSymbolFromReflection( reflection )` returns `undefined`. No idea how to get the internal `node` for such reflections.

Comment (psmyrek) [reply packages/ckeditor5-dev-docs/lib/validators/index.js]: I tried again and this time it worked. Now, all errors found by the validators will be reported along with the node from the source code that has them.

obraz

**Comment by pomek** [comment]
It works as expected. The last thing we need before merging is to cover [684a617](https://github.com/ckeditor/ckeditor5-dev/pull/831/commits/684a6174bf38d43a99c6e8229571ce55c71fde21) (tests).

**Pull request merged**